### PR TITLE
Allow mounting routeless engines with a bound engine name

### LINKED ISSUE #117: Route less engine mounting with a variable
`{{mount someVar}}` does not work where `someVar` is a variable with the engine-name

It throws below error:
`Uncaught Error: Assertion Failed: The first argument of {{mount}} must be quoted, e.g. {{mount "chat-engine"}}.`

Not sure if this constraint is by design or an issue

### PR DESCRIPTION
**Pull request by mike183**
This PR builds upon the work in #14967, introducing the ability to pass bound properties to the `{{mount}}` syntax, enabling the dynamic mounting of "routeless" engines.

This also resolves ember-engines/ember-engines#117

cc/ @rwjblue

### CODE REVIEW

**Comment by rwjblue** [comment]
LGTM

@dgeb / @chancancode / @trentmwillis - Any objections?

File: packages/ember-glimmer/lib/syntax/mount.js [hunk 1]
```diff
@@ -72,23 +67,27 @@ class DynamicEngineReference {
...
 
-    this._lastName = nameOrDef;
-    this._lastDef = new MountDefinition(nameOrDef);
+      this._lastName = nameOrDef;
+      this._lastDef = new MountDefinition(nameOrDef);
 
-    return this._lastDef;
+      return this._lastDef;
+    } else {
+      return null;
```
Comment (trentmwillis) [inline packages/ember-glimmer/lib/syntax/mount.js]: IMO this should log if the argument isn't a `string/undefined/null`.

**Review by trentmwillis — CHANGES_REQUESTED** [review]
Overall, I'm good with it. For bad input though, I'd like to see feedback given to the user.

File: packages/ember-glimmer/lib/syntax/mount.js (hunk 1 above)
Comment (rwjblue) [inline packages/ember-glimmer/lib/syntax/mount.js]: ya, lets have it assert

**Comment by mike183** [comment]
@trentmwillis Thanks for the review, I've added an assertion to check that the provided engine name is valid and also added a test to go along with it.

cc/ @rwjblue

**Review by rwjblue — APPROVED** [review]

**Comment by mike183** [comment]
🎉

**Pull request merged**